### PR TITLE
Update LLVM to 6.0.1 and WebAssembly target refactoring

### PR DESCRIPTION
**Pull request by sbogolepov**
WebAssembly improvements:
* Remove binaryen dependency
* Faster compilation time
* Simplify morecore function
* Simplify launcher.js

### CODE REVIEW

File: konan/konan.properties [hunk 1]
```diff
@@ -54,7 +54,7 @@ osVersionMin.macos_x64 = 10.11
 entrySelector.macos_x64 = -alias _Konan_main _main
 dependencies.macos_x64 = \
     libffi-3.2.1-2-darwin-macos \
-    clang-llvm-5.0.0-darwin-macos
+    clang-llvm-6.0.0-darwin-macos
```
Comment (olonho) [inline konan/konan.properties]: 6.0.1?

File: konan/platforms/zephyr/stm32f4_disco [hunk 2]
```diff
@@ -25,18 +25,18 @@ boardSpecificClangFlags.zephyr_stm32f4_disco = -mabi=aapcs -mthumb -mcpu=cortex-
...
 dependencies.macos_x64-zephyr_stm32f4_disco = \
     gcc-arm-none-eabi-7-2017-q4-major-mac \
-    clang-llvm-5.0.0-darwin-macos \
+    clang-llvm-6.0.0-darwin-macos \
     target-sysroot-2-wasm
 
 targetToolchain.mingw_x64-zephyr_stm32f4_disco = gcc-arm-none-eabi-7-2017-q4-major-win32/arm-none-eabi
 dependencies.mingw_x64-zephyr_stm32f4_disco = \
-    msys2-mingw-w64-x86_64-gcc-7.2.0-clang-llvm-5.0.0-windows-x86-64 \
+    mingw64 \
```
Comment (olonho) [inline konan/platforms/zephyr/stm32f4_disco]: Not a very good name.

File: runtime/src/main/cpp/math/fmodf.cpp [hunk 3]
```diff
@@ -1,6 +1,10 @@
 #include <math.h>
 #include <stdint.h>
 
+#ifdef KONAN_WASM
+#include "../Common.h"
```
Comment (olonho) [inline runtime/src/main/cpp/math/fmodf.cpp]: Please do't do includes by relative paths.
Also add RUNTIME_USED in separate ifdef, code is unreadable this way.

File: konan/platforms/zephyr/stm32f4_disco (hunk 2 above)
Comment (aceisnotmycard) [reply konan/platforms/zephyr/stm32f4_disco]: @msink, it's not ready for use yet :)

Comment (msink) [reply konan/platforms/zephyr/stm32f4_disco]: Will it fix #1734 ?

Comment (aceisnotmycard) [reply konan/platforms/zephyr/stm32f4_disco]: Yes, it should.

**Comment by viteinfinite** [comment]
@olonho Does this allow generating valid bitcode (cf #1564)?
Thank you!

**Comment by sbogolepov** [comment]
@viteinfinite this PR has nothing to do with bitcode embedding.

**Pull request merged**